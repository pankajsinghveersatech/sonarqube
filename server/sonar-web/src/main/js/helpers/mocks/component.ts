/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { ComponentQualifier, TreeComponent } from '../../types/component';
import { MetricKey } from '../../types/metrics';
import { mockMeasureEnhanced } from '../testMocks';

export function mockComponent(overrides: Partial<T.Component> = {}): T.Component {
  return {
    breadcrumbs: [],
    key: 'my-project',
    name: 'MyProject',
    qualifier: ComponentQualifier.Project,
    qualityGate: { isDefault: true, key: '30', name: 'Sonar way' },
    qualityProfiles: [
      {
        deleted: false,
        key: 'my-qp',
        language: 'ts',
        name: 'Sonar way'
      }
    ],
    tags: [],
    ...overrides
  };
}

export function mockTreeComponent(overrides: Partial<TreeComponent>): TreeComponent {
  return {
    key: 'my-key',
    qualifier: ComponentQualifier.Project,
    name: 'component',
    visibility: 'public',
    ...overrides
  };
}

export function mockComponentMeasure(
  file = false,
  overrides: Partial<T.ComponentMeasure> = {}
): T.ComponentMeasure {
  if (file) {
    return {
      key: 'foo:src/index.tsx',
      name: 'index.tsx',
      qualifier: ComponentQualifier.File,
      path: 'src/index.tsx',
      measures: [{ metric: MetricKey.bugs, value: '1', bestValue: false }],
      ...overrides
    };
  }
  return {
    key: 'foo',
    name: 'Foo',
    qualifier: ComponentQualifier.Project,
    measures: [{ metric: MetricKey.bugs, value: '12', bestValue: false }],
    ...overrides
  };
}

export function mockComponentMeasureEnhanced(
  overrides: Partial<T.ComponentMeasureEnhanced> = {}
): T.ComponentMeasureEnhanced {
  return {
    ...mockComponentMeasure(false, overrides as T.ComponentMeasure),
    leak: undefined,
    measures: [mockMeasureEnhanced()],
    value: undefined,
    ...overrides
  };
}
