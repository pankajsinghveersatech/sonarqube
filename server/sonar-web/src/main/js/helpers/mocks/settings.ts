/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { Setting, SettingCategoryDefinition, SettingWithCategory } from '../../types/settings';

export function mockDefinition(
  overrides: Partial<SettingCategoryDefinition> = {}
): SettingCategoryDefinition {
  return {
    key: 'foo',
    category: 'foo category',
    fields: [],
    options: [],
    subCategory: 'foo subCat',
    ...overrides
  };
}

export function mockSetting(overrides: Partial<Setting> = {}): Setting {
  return {
    key: 'foo',
    value: '42',
    inherited: true,
    definition: {
      key: 'foo',
      name: 'Foo setting',
      description: 'When Foo then Bar',
      type: 'INTEGER',
      options: []
    },
    ...overrides
  };
}

export function mockSettingWithCategory(
  overrides: Partial<SettingWithCategory> = {}
): SettingWithCategory {
  return {
    key: 'foo',
    value: '42',
    inherited: true,
    definition: {
      key: 'foo',
      name: 'Foo setting',
      description: 'When Foo then Bar',
      type: 'INTEGER',
      options: [],
      category: 'general',
      fields: [],
      subCategory: 'email'
    },
    ...overrides
  };
}
