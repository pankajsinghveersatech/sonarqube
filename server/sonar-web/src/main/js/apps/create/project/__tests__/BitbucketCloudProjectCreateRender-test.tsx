/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { shallow } from 'enzyme';
import * as React from 'react';
import { mockBitbucketCloudAlmSettingsInstance } from '../../../../helpers/mocks/alm-settings';
import BitbucketCloudProjectCreateRenderer, {
  BitbucketCloudProjectCreateRendererProps
} from '../BitbucketCloudProjectCreateRender';

it('Should render correctly', () => {
  expect(shallowRender()).toMatchSnapshot();
  expect(shallowRender({ settings: undefined })).toMatchSnapshot('Wrong config');
  expect(shallowRender({ loading: true })).toMatchSnapshot('Loading...');
  expect(
    shallowRender({
      showPersonalAccessTokenForm: true
    })
  ).toMatchSnapshot('Need App password');
});

function shallowRender(props?: Partial<BitbucketCloudProjectCreateRendererProps>) {
  return shallow(
    <BitbucketCloudProjectCreateRenderer
      onImport={jest.fn()}
      isLastPage={true}
      loading={false}
      loadingMore={false}
      onLoadMore={jest.fn()}
      onPersonalAccessTokenCreated={jest.fn()}
      onSearch={jest.fn()}
      resetPat={false}
      searching={false}
      searchQuery=""
      settings={mockBitbucketCloudAlmSettingsInstance()}
      showPersonalAccessTokenForm={false}
      {...props}
    />
  );
}
