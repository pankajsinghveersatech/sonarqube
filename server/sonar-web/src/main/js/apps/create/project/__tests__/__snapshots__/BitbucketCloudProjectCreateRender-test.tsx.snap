// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly 1`] = `
<Fragment>
  <CreateProjectPageHeader
    title={
      <span
        className="text-middle"
      >
        <img
          alt=""
          className="spacer-right"
          height="24"
          src="/images/alm/bitbucket.svg"
        />
        onboarding.create_project.bitbucketcloud.title
      </span>
    }
  />
  <BitbucketCloudSearchForm
    isLastPage={true}
    loadingMore={false}
    onImport={[MockFunction]}
    onLoadMore={[MockFunction]}
    onSearch={[MockFunction]}
    searchQuery=""
    searching={false}
  />
</Fragment>
`;

exports[`Should render correctly: Loading... 1`] = `
<Fragment>
  <CreateProjectPageHeader
    title={
      <span
        className="text-middle"
      >
        <img
          alt=""
          className="spacer-right"
          height="24"
          src="/images/alm/bitbucket.svg"
        />
        onboarding.create_project.bitbucketcloud.title
      </span>
    }
  />
  <i
    className="spinner"
  />
</Fragment>
`;

exports[`Should render correctly: Need App password 1`] = `
<Fragment>
  <CreateProjectPageHeader
    title={
      <span
        className="text-middle"
      >
        <img
          alt=""
          className="spacer-right"
          height="24"
          src="/images/alm/bitbucket.svg"
        />
        onboarding.create_project.bitbucketcloud.title
      </span>
    }
  />
  <PersonalAccessTokenForm
    almSetting={
      Object {
        "alm": "bitbucketcloud",
        "key": "key",
      }
    }
    onPersonalAccessTokenCreated={[MockFunction]}
    resetPat={false}
  />
</Fragment>
`;

exports[`Should render correctly: Wrong config 1`] = `
<Fragment>
  <CreateProjectPageHeader
    title={
      <span
        className="text-middle"
      >
        <img
          alt=""
          className="spacer-right"
          height="24"
          src="/images/alm/bitbucket.svg"
        />
        onboarding.create_project.bitbucketcloud.title
      </span>
    }
  />
  <WrongBindingCountAlert
    alm="bitbucketcloud"
    canAdmin={false}
  />
</Fragment>
`;
